gnd/src/test/java/com/google/android/gnd/persistence/local/LocalDataStoreTest.java

```diff
@@ -352,136 +341,75 @@ public void testMergeFeature() {
     localDataStore.applyAndEnqueue(mutation).test().assertComplete();
 
     ImmutableList<Mutation> savedMutations =
         localDataStore.getPendingMutations(mutation.getFeatureId()).blockingGet();
     Assert.assertEquals(2, savedMutations.size());
-
     // ignoring the first item, which is a FeatureMutation. Already tested separately.
-    ObservationMutation savedMutation = (ObservationMutation) savedMutations.get(1);
-    Assert.assertEquals(mutation.getResponseDeltas(), savedMutation.getResponseDeltas());
-    Assert.assertEquals(mutation.getType(), savedMutation.getType());
-    Assert.assertEquals(mutation.getUserId(), savedMutation.getUserId());
-    Assert.assertEquals(mutation.getProjectId(), savedMutation.getProjectId());
-    Assert.assertEquals(mutation.getFeatureId(), savedMutation.getFeatureId());
-    Assert.assertEquals(layer.getId(), savedMutation.getLayerId());
-    Assert.assertEquals(mutation.getClientTimestamp(), savedMutation.getClientTimestamp());
-    Assert.assertEquals(0, savedMutation.getRetryCount());
-    Assert.assertNull(savedMutation.getLastError());
+    assertObservationMutation(mutation, (ObservationMutation) savedMutations.get(1));
 
     // check if the observation was saved properly to local database
-    Feature feature = localDataStore.getFeature(project, mutation.getFeatureId()).blockingGet();
+    Feature feature =
+        localDataStore.getFeature(TEST_PROJECT, mutation.getFeatureId()).blockingGet();
     Observation observation =
         localDataStore.getObservation(feature, mutation.getObservationId()).blockingGet();
-    Assert.assertEquals(mutation.getObservationId(), observation.getId());
-    Assert.assertEquals(user, observation.getCreated().getUser());
-    Assert.assertEquals(feature, observation.getFeature());
-    Assert.assertEquals(layer.getForm().get(), observation.getForm());
-    Assert.assertEquals(project, observation.getProject());
-    Assert.assertEquals(user, observation.getLastModified().getUser());
-    MatcherAssert.assertThat(
-        ResponseMap.builder().applyDeltas(mutation.getResponseDeltas()).build(),
-        samePropertyValuesAs(observation.getResponses()));
+    assertObservation(mutation, observation);
 
     // now update the inserted observation with new responses
     ImmutableList<ResponseDelta> deltas =
-        ImmutableList.<ResponseDelta>builder()
-            .add(
-                ResponseDelta.builder()
-                    .setFieldId("really new field")
-                    .setNewResponse(TextResponse.fromString("value for the really new field"))
-                    .build())
-            .build();
+        ImmutableList.of(
+            ResponseDelta.builder()
+                .setFieldId("really new field")
+                .setNewResponse(TextResponse.fromString("value for the really new field"))
+                .build());
     mutation = mutation.toBuilder().setResponseDeltas(deltas).setType(Mutation.Type.UPDATE).build();
```

Comment:
Please replace secondary assertions (i.e. those that don't test the behavior in question) with `blocking*().` If those code paths don't already have their own tests, please be sure to add them.